.gitignore

```diff
@@ -1,2 +1,11 @@
+
+#NPM
 node_modules/
+
+#IDE Specific
 .idea
+
+#Module specific build artefacts
```

Comment:
I think we should list one by one the files that are actually generated by the gulp task here, because we may include other files in the future that can fulfill these patterns.